src/main/scala/com/crealytics/spark/excel/ExcelRelation.scala

```diff
@@ -25,7 +26,10 @@ case class ExcelRelation(
   )
   (@transient val sqlContext: SQLContext)
 extends BaseRelation with TableScan with PrunedScan {
-  val workbook = WorkbookFactory.create(new FileInputStream(location))
+  val path = new Path(location)
+  val inputStream = FileSystem.get(path.toUri, sqlContext.sparkContext.hadoopConfiguration).open(path)
+  val workbook = WorkbookFactory.create(inputStream)
+  inputStream.close()
```

Comment:
Oh, I didn't even think about closing the stream :wink:
Did you make sure that `WorkbookFactory.create` reads all data eagerly?
Otherwise we'd get into trouble later when it tries to read stuff from the already closed stream.